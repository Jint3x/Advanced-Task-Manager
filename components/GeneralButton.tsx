
interface GeneralButton_I {
  text: string,
  customClasses: string,
  clickFunction: Function
}

export default function GeneralButton({text, clickFunction, customClasses}: GeneralButton_I) {
  return (
    <button
     onClick={() => clickFunction() }
     className={`rounded-[10px] text-base md:text-xl " ${customClasses}`}
    >
      {text}
    </button>
  )
}