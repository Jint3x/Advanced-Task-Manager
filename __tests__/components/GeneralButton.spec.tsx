import GeneralButton from "../../components/GeneralButton";
import { screen, render } from "@testing-library/react";
import renderer from 'react-test-renderer';

describe("Tests component: <GeneralButton />", () => {
  it("Calls a function that is passed down to the button", () => {
    let text = "Click me!";
    let customClasses = "";
    let clickFunction = jest.fn(() => null);

    render(
      <GeneralButton 
       text={text}
       customClasses={customClasses}
       clickFunction={clickFunction}
      />
    )

    let btn = screen.getByText(text);
    btn.click();

    expect(clickFunction.mock.calls.length).toBe(1);
  })


  it("Applies the provided classes to the button & verifies that the element is a button", () => {
    let text = "Click me!";
    let customClasses = "class1 class2 class3";
    let clickFunction = jest.fn(() => null);

    render(
      <GeneralButton 
       text={text}
       customClasses={customClasses}
       clickFunction={clickFunction}
      />
    )

    let btn = screen.getByText(text);
    let btnClasses = Object.values(btn.classList);
    let appliedClasses = customClasses.split(" ");

    for (let appliedClass of appliedClasses) {
      expect(btnClasses).toContain(appliedClass);
    };

    expect(btn.nodeName).toBe("BUTTON")
  })


  it("Takes a snapshot of the button", () => {
    let text = "Click me!";
    let customClasses = "class1 class2 class3";
    let clickFunction = jest.fn(() => null);

    const tree = renderer
    .create(
      <GeneralButton 
       text={text}
       customClasses={customClasses}
       clickFunction={clickFunction}
      />
    )
    .toJSON();

    expect(tree).toMatchSnapshot()
  })
})