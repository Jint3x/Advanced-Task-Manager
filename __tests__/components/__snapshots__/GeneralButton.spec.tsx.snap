// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests component: <GeneralButton /> Takes a snapshot of the button 1`] = `
<button
  className="rounded-[10px] text-base md:text-xl \\" class1 class2 class3"
  onClick={[Function]}
>
  Click me!
</button>
`;
